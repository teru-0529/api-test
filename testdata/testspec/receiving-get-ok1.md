# 受注取得(OK case1)

受注情報の取得処理（成功ケース）。

## テスト対象DBの初期化

```http
# @name sdequenceリセット
POST {{$dotenv DB_RESETER_API_HOST}}/schemas/orders/action-seq-reset HTTP/1.1
Content-Type: application/json

[
  "order_no_seed",
  "product_id_seed"
]
```

```http
# @name table初期化
POST {{$dotenv DB_RESETER_API_HOST}}/schemas/orders/action-truncate HTTP/1.1
Content-Type: application/json

[
  "products",
  "receivings"
]
```

## テストデータ登録

```http
# @name bulk insert(orders.products)
POST {{$dotenv POSTGREST_API_HOST}}/products HTTP/1.1
Content-Type: application/json
Content-Profile: orders

[
  {
    "cost_price": 20000,
    "product_name": "日本刀"
  },
  {
    "cost_price": 40000,
    "product_name": "火縄銃"
  },
  {
    "cost_price": 15000,
    "product_name": "弓"
  }
]
```

```http
# @name bulk insert(orders.receivings)
POST {{$dotenv POSTGREST_API_HOST}}/receivings HTTP/1.1
Content-Type: application/json
Content-Profile: orders

[
  {
    "customer_name": "徳川商事株式会社",
    "operator_name": "織田信長"
  },
  {
    "customer_name": "株式会社武田物産",
    "operator_name": "豊臣秀吉"
  }
]
```

```http
# @name bulk insert(orders.receiving_details)
POST {{$dotenv POSTGREST_API_HOST}}/receiving_details HTTP/1.1
Content-Type: application/json
Content-Profile: orders

[
  {
    "order_no": "RO-0000001",
    "product_id": "P0002",
    "receiving_quantity": 5,
    "sellling_price": 58000
  },
  {
    "order_no": "RO-0000002",
    "product_id": "P0001",
    "receiving_quantity": 3,
    "sellling_price": 32000
  },
  {
    "order_no": "RO-0000002",
    "product_id": "P0003",
    "receiving_quantity": 1,
    "sellling_price": 19000
  }
]
```

## テスト実行

```http
# @name execute API
GET {{$dotenv ORDERS_API_HOST}}/receivings/RO-0000002 HTTP/1.1
Content-Type: application/json
x-account-id: P0673822
```

## 検証対象

- API実行結果の http status が(200)であること
- API実行結果が正しいこと

-----
api-test-specification.(v1.0.0)
