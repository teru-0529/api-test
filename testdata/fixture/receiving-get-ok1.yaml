dataType: api_test_fixture
version: 1.0.0
# ----+----+----+
# 基本情報
# name: 受注取得(OK case1)
description: 受注情報の取得処理（成功ケース）。
# ----+----+----+
# 情報クリア
reset:
  sequences:
    - schema: orders
      items: [order_no_seed, product_id_seed]
  tables:
    - schema: orders
      items: [products, receivings]
# ----+----+----+
# 事前処理
setupTable:
  - schema: orders
    table: products
    body: '[
      {"product_name": "日本刀","cost_price": 20000},
      {"product_name": "火縄銃","cost_price": 40000},
      {"product_name": "弓","cost_price": 15000}
      ]'
  - schema: orders
    table: receivings
    body: '[
      {"operator_name": "織田信長","customer_name": "徳川商事株式会社"},
      {"operator_name": "豊臣秀吉","customer_name": "株式会社武田物産"}
      ]'
  - schema: orders
    table: receiving_details
    body: '[
      {"order_no": "RO-0000001","product_id": "P0002","receiving_quantity": 5,"sellling_price": 58000},
      {"order_no": "RO-0000002","product_id": "P0001","receiving_quantity": 3,"sellling_price": 32000},
      {"order_no": "RO-0000002","product_id": "P0003","receiving_quantity": 1,"sellling_price": 19000}
      ]'
# API実行
execute:
  hostKey: ORDERS_API_HOST
  method: GET
  path: /receivings/RO-0000002
  headers:
    - key: x-account-id
      value: P0673822
  body: null
# ----+----+----+
# 検証
verification:
  httpStatus: 200
  execResult:
    check: true
    exclude: []
    # exclude: [/orderDate]
  tables: []
