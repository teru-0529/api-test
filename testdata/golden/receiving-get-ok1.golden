{
  "customerName": "株式会社武田物産",
  "details": [
    {
      "cancelQuantity": 0,
      "costPrice": 20000,
      "orderQuantity": 3,
      "orderStatus": "WORK_IN_PROGRESS",
      "productId": "P0001",
      "profitRate": 0.38,
      "remainingQuantity": 3,
      "sellingPrice": 32000,
      "shippingQuantity": 0
    },
    {
      "cancelQuantity": 0,
      "costPrice": 15000,
      "orderQuantity": 1,
      "orderStatus": "WORK_IN_PROGRESS",
      "productId": "P0003",
      "profitRate": 0.21,
      "remainingQuantity": 1,
      "sellingPrice": 19000,
      "shippingQuantity": 0
    }
  ],
  "operatorName": "豊臣秀吉",
  "orderNo": "RO-0000002",
  "orderStatus": "WORK_IN_PROGRESS",
  "remainingOrderPrice": 115000,
  "totalOrderPrice": 115000
}